src/wallet/wallet2.cpp

```diff
@@ -4733,19 +4503,24 @@ uint64_t wallet2::unlocked_dust_balance(const tx_dust_policy &dust_policy) const
 //----------------------------------------------------------------------------------------------------
 void wallet2::get_hard_fork_info(uint8_t version, uint64_t &earliest_height)
 {
-  boost::optional<std::string> result = m_node_rpc_proxy.get_earliest_height(version, earliest_height);
-  throw_on_rpc_response_error(result, "get_hard_fork_info");
+  cryptonote::rpc::hard_fork_info info;
+
+  bool r = m_daemon.hardForkInfo(version, info);
+  CHECK_AND_ASSERT_THROW_MES(r, "Failed to get hard fork status");
+
+  earliest_height = info.earliest_height;
```

Comment:
Why removed the proxy ? If 0MQ is significantly faster then JSON RPC, then it might be OK, but it'd need to be checked.